libsnark/gadgetlib1/gadgets/hashes/anemoi/anemoi_components.hpp

```diff
@@ -0,0 +1,114 @@
+/** @file
+ *****************************************************************************
+
+ Declaration of interfaces for top-level SHA256 gadgets.
+
+ *****************************************************************************
+ * @author     This file is part of libsnark, developed by SCIPR Lab
+ *             and contributors (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef LIBSNARK_GADGETLIB1_GADGETS_HASHES_ANEMOI_COMPONENTS_HPP_
+#define LIBSNARK_GADGETLIB1_GADGETS_HASHES_ANEMOI_COMPONENTS_HPP_
+
+#include <libsnark/common/data_structures/merkle_tree.hpp>
+#include <libsnark/gadgetlib1/gadgets/basic_gadgets.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/hash_io.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/sha256/sha256_components.hpp>
+
+namespace libsnark
+{
+
+/// Compute y = alpha x^2 + beta
+/// x: input
+/// y: output
+/// a,b: intermediate values
+/// alpha, beta: constants
+template<typename FieldT> class anemoi_power_two_gadget : public gadget<FieldT>
+{
+private:
+    // intermediate values
+    pb_variable<FieldT> a;
+    pb_variable<FieldT> b;
+
+public:
+    // input
+    const pb_variable<FieldT> x;
+    // output
+    const pb_variable<FieldT> y;
+    // constants
+    pb_variable<FieldT> alpha;
+    pb_variable<FieldT> beta;
+
+    anemoi_power_two_gadget(
+        protoboard<FieldT> &pb,
+        const pb_variable<FieldT> &x,
+        const pb_variable<FieldT> &y,
+        const pb_variable<FieldT> &alpha,
```

Comment:
Are these genuinely variables? No problem if they are, but I would expect them to be baked into the circuit.

Reply:
To be handled in #76